typescript/cli/src/config/warp.ts

```diff
@@ -0,0 +1,126 @@
+import { confirm, input } from '@inquirer/prompts';
+import { ethers } from 'ethers';
+import { z } from 'zod';
+
+import { TokenType } from '@hyperlane-xyz/hyperlane-token';
+
+import { errorRed, logBlue, logGreen } from '../../logger.js';
+import {
+  runMultiChainSelectionStep,
+  runSingleChainSelectionStep,
+} from '../utils/chains.js';
+import { FileFormat, readYamlOrJson, writeYamlOrJson } from '../utils/files.js';
+
+import { readChainConfigIfExists } from './chain.js';
+
+const ConnectionConfigSchema = {
+  mailbox: z.string().optional(),
+  interchainGasPaymaster: z.string().optional(),
+  interchainSecurityModule: z.string().optional(),
+  foreignDeployment: z.string().optional(),
+};
```

Comment:
can we not reuse SDK types here somehow?
z.object() or something?

Reply:
https://zod.dev/?id=custom-schemas maybe this?

Reply:
Hmm, I think that's something different. I'm skeptical that a lib can generate runtime code from a TS type. Keep in mind types get dissolved again during compilation.